client/src/controllers/CountController.ts

```diff
@@ -0,0 +1,55 @@
+import { Controller } from '@hotwired/stimulus';
+import { ngettext } from '../utils/gettext';
+
+export class CountController extends Controller<HTMLFormElement> {
+  static classes = ['active'];
+  static targets = ['label', 'total'];
+  static values = {
+    container: { type: String, default: 'body' },
+    find: {
+      type: String,
+      default: '.error-message,.help-critical',
+    },
+    total: { type: Number, default: 0 },
+  };
+
+  declare containerValue: string;
+  declare findValue: string;
+  declare totalValue: number;
+
+  declare readonly activeClass: string;
+  declare readonly hasActiveClass: boolean;
+  declare readonly hasLabelTarget: boolean;
+  declare readonly hasTotalTarget: boolean;
+  declare readonly labelTarget: HTMLElement;
+  declare readonly totalTarget: HTMLElement;
+
+  connect() {
+    this.count();
+  }
+
+  count() {
+    const containerElement = document.querySelector(
+      this.containerValue || 'body',
+    );
+    if (containerElement) {
+      this.totalValue = containerElement.querySelectorAll(
+        this.findValue,
+      ).length;
+    }
+  }
+
+  totalValueChanged(total) {
+    if (this.hasActiveClass) {
+      this.element.classList.toggle(this.activeClass, total > 0);
+    }
+    if (this.hasLabelTarget && total > 0) {
+      this.totalTarget.innerHTML = total;
+      this.labelTarget.innerHTML = ngettext(
```

Comment:
I will pull this label out to a new method so we can make this a bit easier to override with custom labels in the future.